### PR TITLE
Modulemap refactor

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
fixed conflicts

File: Sources/PackageLoading/ModuleMapGeneration.swift [hunk 1]
```diff
@@ -40,8 +40,25 @@ extension ClangModule: ModuleMapProtocol {
...
+///
+/// Modulemap is generated under the following rules provided it is not already present in include directory:
+///
+/// * "include/foo/foo.h" exists and `foo` is the only directory under include directory.
+///    Generates: `umbrella header "/path/to/include/foo/foo.h"`
+/// * "include/foo.h" exists and include contains no other directory.
+///    Generates: `umbrella header "/path/to/include/foo.h"`
+/// *  Otherwise in all other cases.
+///    Generates: `umbrella "path/to/include"`
+public struct ModuleMapGenerator {
```
Comment (ddunbar) [inline Sources/PackageLoading/ModuleMapGeneration.swift]: At this point file should probably be `ModuleMapGenerator`, I guess?

File: Sources/PackageLoading/ModuleMapGeneration.swift [hunk 2]
```diff
@@ -55,8 +55,12 @@ public struct ModuleMapGenerator {
     /// The clang module to operate on.
     private let module: ClangModule
 
-    public init(for module: ClangModule) {
+    /// The file system to be used.
+    private var fileSystem: FileSystem
```
Comment (ddunbar) [inline Sources/PackageLoading/ModuleMapGeneration.swift]: I think I have realized that mutability does work for `FileSystem` and am going to remove it, but I haven't made up my mind completely. If we do, this var will go away (this is just a note to myself, I guess).

**Comment by ddunbar** [comment]
This is great, LGTM!

File: Sources/PackageLoading/ModuleMapGeneration.swift (hunk 1 above)
Comment (aciidb0mb3r) [inline Sources/PackageLoading/ModuleMapGeneration.swift]: Right, done

**Pull request merged**